test/cli/ui/work_queue_test.rb

```diff
@@ -0,0 +1,150 @@
+# typed: false
+
+require 'test_helper'
+require 'cli/ui/work_queue'
+
+module CLI
+  module UI
+    class WorkQueueTest < Minitest::Test
```

Comment:
the sleeps in these tests feel a bit fragile and prone to flakiness. could we do something like use queues to get the threads into the exact states we expect? e.g. in `test_future_value_blocks_until_result_available` maybe we could do something like
* main_thread: start the work thread
* main_thread: block waiting on a signal that work_thread is up
* work_thread: signal that it's up
* work_thread: block waiting on a signal to shut down
* main_thread: check `value_thread.alive?`
* main_thread: signal work_thread to shut down